pkg/util/gce/gce.go

```diff
@@ -83,6 +83,19 @@ func GetClusterName() (string, error) {
 	return clusterName, nil
 }
 
+// GetPublicIPv4 returns the public IPv4 address of the current GCE instance
+func GetPublicIPv4() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
+		return "", fmt.Errorf("cloud provider is disabled by configuration")
+	}
+	clusterName, err := getResponseWithMaxLength(metadataURL+"/instance/network-interfaces/0/access-configs/0/external-ip",
```

Comment:
May want to make the whole URL a constant as well.

Reply:
I'm mimicking how it is done with the other endpoints, I don't know that a constant would help much here.